shared/actions/platform.specific.native.js

```diff
@@ -72,8 +72,8 @@ function configurePush () {
           type: 'push:error',
         }: PushConstants.PushError))
       },
-      // Don't request permissions now, we'll ask later, after showing UI
-      requestPermissions: false,
+      // Don't request permissions for ios, we'll ask later, after showing UI
+      requestPermissions: !isIOS,
```

Comment:
If this is false, then the library won't try to do the registration flow until you explicitly ask PushNotifications.requestPermissions.